ignite/pkg/cosmosgen/generate.go

```diff
@@ -105,22 +119,128 @@ func (g *generator) setup() (err error) {
 				return err
 			}
 
+			var includes []string
+			if len(modules) > 0 {
+				// For versioning issues, we do dependency/includes resolution per module
+				includes, err = g.resolveIncludes(path)
+				if err != nil {
+					return err
+				}
+			}
+
 			modulesInPath = ModulesInPath{
-				Path:    path,
-				Modules: modules,
+				Path:     path,
+				Modules:  modules,
+				Includes: includes,
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = append(g.thirdModules[modulesInPath.Path], modulesInPath.Modules...)
+		g.thirdModules[modulesInPath.Path] = append(
+			g.thirdModules[modulesInPath.Path],
+			modulesInPath.Modules...,
+		)
+
+		if modulesInPath.Includes != nil {
+			g.thirdModuleIncludes[modulesInPath.Path] = append(
+				g.thirdModuleIncludes[modulesInPath.Path],
+				modulesInPath.Includes...,
+			)
+		}
 	}
 
 	return nil
 }
 
+func (g *generator) getProtoIncludeFolders(modPath string) []string {
+	// Add default protoDir and default includeDirs
+	includePaths := []string{filepath.Join(modPath, g.protoDir)}
+	for _, dir := range g.o.includeDirs {
+		includePaths = append(includePaths, filepath.Join(modPath, dir))
+	}
+	return includePaths
+}
+
+func (g *generator) findBufPath(modpath string) (string, error) {
+	var bufPath string
+	err := filepath.WalkDir(modpath, func(path string, _ fs.DirEntry, err error) error {
```

Comment:
do we need to use the walk? the `buf.yaml` is always in the root of the project

Reply:
not necessarily...this is looking at a go module path in the dependencies....usually it's under proto/buf.yaml but that's just convention, not enforced